plugins/continuous-toolbox/src/ContinuousFlyout.js

```diff
@@ -190,69 +190,52 @@ export class ContinuousFlyout extends Blockly.VerticalFlyout {
     return 0;
   }
 
-  /**
-   * Overrides the position function solely to change the x coord in RTL mode.
-   * The base function allows the workspace to go "under" the flyout, so
-   * to calculate the left edge of the flyout in RTL you would just subtract
-   * the flyout width from the total viewWidth to get x. However, in our
-   * flyout, the workspace already starts at the left edge of the flyout, so
-   * we don't need to subtract the flyout width again.
-   * Ideally there would be a smaller method for us to override instead,
-   * but for now we copy/paste this method and make our fixes.
-   * @override
-   */
-  position() {
+  /** @override */
+  getX() {
     if (!this.isVisible()) {
-      return;
+      return 0;
     }
-    const targetWorkspaceMetrics = this.targetWorkspace.getMetrics();
-    if (!targetWorkspaceMetrics) {
-      // Hidden components will return null.
-      return;
-    }
-    // Record the height for Blockly.Flyout.getMetrics_
-    this.height_ = targetWorkspaceMetrics.viewHeight;
-
-    const edgeWidth = this.width_ - this.CORNER_RADIUS;
-    const edgeHeight =
-        targetWorkspaceMetrics.viewHeight - 2 * this.CORNER_RADIUS;
-    this.setBackgroundPath_(edgeWidth, edgeHeight);
-
-    // Y is always 0 since this is a vertical flyout.
-    const y = 0;
+    const metricsManager = this.targetWorkspace.getMetricsManager();
+    const absoluteMetrics = metricsManager.getAbsoluteMetrics();
+    const viewMetrics = metricsManager.getViewMetrics();
+    const toolboxMetrics = metricsManager.getToolboxMetrics();
     let x = 0;
-    // If this flyout is the toolbox flyout.
-    if (this.targetWorkspace.toolboxPosition == this.toolboxPosition_) {
+
+    // If this flyout is not the trashcan flyout (e.g. toolbox or mutator).
+    // Trashcan flyout is opposite the main flyout.
+    if (this.targetWorkspace.toolboxPosition === this.toolboxPosition_) {
       // If there is a category toolbox.
-      if (targetWorkspaceMetrics.toolboxWidth) {
-        if (this.toolboxPosition_ == Blockly.TOOLBOX_AT_LEFT) {
-          x = targetWorkspaceMetrics.toolboxWidth;
+      // Simple (flyout-only) toolbox.
+      if (this.targetWorkspace.getToolbox()) {
+        if (this.toolboxPosition_ === Blockly.utils.toolbox.Position.LEFT) {
+          x = toolboxMetrics.width;
         } else {
-          // TODO(https://github.com/google/blockly/issues/4396): Use a better
-          // API to adjust this value.
-          // This is the only line that changed from the original.
-          x = targetWorkspaceMetrics.viewWidth;
+          // This is the one thing that has been changed from the base
+          // implementation, to account for RTL. This makes it so blocks cannot
+          // go under the flyout in RTL mode.
+          x = viewMetrics.width;
```

Comment:
If this is the only thing that is changed, couldn't the function be rewritten as:

```JS
  getX() {
    if (this.isVisible() &&
        this.targetWorkspace.toolboxPosition === this.toolboxPosition_ &&
        this.targetWorkspace.getToolbox() &&
        this.toolboxPosition_ !== Blockly.utils.toolbox.Position.LEFT) {
      // This is the one thing that has been changed from the base
      // implementation, to account for RTL. This makes it so blocks cannot
      // go under the flyout in RTL mode.
      return viewMetrics.width;
    }
    return super.getX();
}
```

(I'm not certain whether the call to `getToolbox()` is strictly necessary, but have preserved it for compatibility.)

That would seem to be at least a little more robust against getting out of sync with changes to the superclass implementation.

Reply:
Yep that's much better!